OpenRA.Game/Traits/World/ScreenMap.cs

```diff
@@ -88,6 +91,41 @@ public void Remove(Actor a)
 			partitionedActors.Remove(a);
 		}
 
+		public void Add(IEffect effect, WPos position, Size size)
+		{
+			var screenPos = worldRenderer.ScreenPxPosition(position);
+			var screenBounds = new Rectangle(screenPos.X - size.Width / 2, screenPos.Y - size.Height / 2, size.Width, size.Height);
+			partitionedEffects.Add(effect, screenBounds);
+		}
+
+		public void Add(IEffect effect, WPos position, Sprite sprite)
+		{
+			var size = new Size((int)sprite.Size.X, (int)sprite.Size.Y);
+			Add(effect, position, size);
+		}
+
+		public void Update(IEffect effect, WPos position, Size size)
+		{
+			if (!partitionedEffects.Contains(effect))
+				return;
```

Comment:
Given we don't check this for actors/frozen actors (i.e. we crash if you call Update without being Added first), should we just remove this check for consistency?

Reply:
That might actually make sense. Silently skipping effects that weren't added just increases the chance of buggy `ISpatiallyPartitionedEffect` implementations not getting caught immediately.